### PR TITLE
Engine/sprite atlas

### PR DESCRIPTION
**Pull request by UbiBelETF**
Added support for tiled spritesheets. The format for the spritesheet is a simple custom textual format and not json, for simplicity. The format can contain the following lines:

- <blank> (skipped in parsing)
- NAME X Y W H (builds a frame called NAME, starting at (X, Y) with dimensions (W, H)
- NAME X Y W H A (builds A frames called NAME:N, where N is between 1 and A, from (X + W * (N - 1), Y) with dimensions (W, H), used for creating multiple frames of the same dimensions, in the same row, under the "same" name)

The texture names generated from this file always start with "spritesheets:SPRITESHEET_NAME:NAME".

The system for parsing spritesheets is done in `sprite_render.cpp`, but adding this system requires a lot of changes elsewhere.

### CODE REVIEW

File: source/dagger/core/graphics/sprite.h [hunk 1]
```diff
@@ -21,8 +37,16 @@ namespace dagger
 		ViewPtr<Shader> shader{ Shader::s_FirstLoadedShader };
 	};
 
-	void AssignSpriteTexture(Sprite& spriteTarget_, String textureName_);
-	void AssignSpriteTexture(Sprite& spriteTarget_, ViewPtr<Texture> texture_);
+	struct Spritesheet
```
Comment (DPrivalov) [inline source/dagger/core/graphics/sprite.h]: I don't have a proper name for this, but seems like Spritesheet is a bit misleading

File: source/dagger/main.cpp [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 #include "core/engine.h"
-#include "gameplay/ping_pong/ping_pong_main.h"
-#include "gameplay/platformer/platformer_main.h"
-#include "gameplay/racing/racing_main.h"
-#include "gameplay/team_game/team_game_main.h"
+//#include "gameplay/ping_pong/ping_pong_main.h"
+//#include "gameplay/platformer/platformer_main.h"
+//#include "gameplay/racing/racing_main.h"
+//#include "gameplay/team_game/team_game_main.h"
+#include "gameplay/tiles_example/tiles_example_main.h"
```
Comment (DPrivalov) [inline source/dagger/main.cpp:7]: why do u need to comment everything else here? 
it'll make enabling of other examples a bit harder

**Review by DPrivalov — APPROVED** [review]

**Review by SofijaBosk — APPROVED** [review]

**Review by milosmilosevic13 — APPROVED** [review]

File: source/dagger/core/graphics/sprite.h (hunk 1 above)
Comment (UbiBelETF) [reply source/dagger/core/graphics/sprite.h]: Agreed! Let's think about this collectively :D

File: source/dagger/main.cpp (hunk 2 above)
Comment (UbiBelETF) [reply source/dagger/main.cpp:7]: I just wanted to get the game to load faster :D we can get this out, really

File: source/dagger/core/graphics/sprite.h (hunk 1 above)
Comment (UbiBelETF) [reply source/dagger/core/graphics/sprite.h]: SpriteFrame!

**Pull request merged**